### PR TITLE
VerifyScript: get orders only for the brackets, and not all orders

### PR DESCRIPTION
**Pull request by josojo**
[dependend on Bens PR https://github.com/gnosis/dex-liquidity-provision/pull/220]

Since it takes quite long to get all orders, I changed the script to retrieve only the orders of brackets to be checked.

testplan;
```
npx truffle exec scripts/verify_correctness.js --masterSafe=0x5CC3a4C846aB8b0B78a729b8B1a50e2E0BB66740 --brackets=0x7c46953a4f0b404e21EBC0d69883aDAef5AAa5C5,0x41B77D6fAB22F67595dAF54C21b8Fc1b3812b528,0xd349236D1333F68761bbE263Af02D57B4aC928da,0x0F5D2E870f5d638D460F128530E278f0C7e4E2C6,0x5C111aFFdA2b8de51DDB7a2111111d70Ee494995,0x10aDB8e1FcDAe35123A7587e9A0E8e18d5445DD0,0xF8a02315A2eDf95d28AF945264434371fBFd43F8,0xeE223cF4292ac20533e7A0b81c1D4FEeD6eCf38C,0xe31F1C4e3B9699b971b7D4e7a8FC4eF0E39a70F9,0x2A0E02dd94c2b34ba8f94eB00807A3ABa34e62cD,0x5a80263780898C6AF01A6d57FA717aB8E3640bc3,0x47628ad01892a06cA574e45dfe8Bfb69a1cB1971,0x1c97057713C2C56599f694A920b060f1da5D94A7,0x364Ad04e974A7CbCbA5C07A867edafff5Dd12586,0xda43a32F07A41B0e1b6b53a62B593C577c65C81a,0xc31997f8e495a51E3D2346674d0Edb6804dEc1D3,0xd07C67bDf9454B6Ea26b2f5E1160A5EEE1b0f299,0xc8E1275babc318721EAa73a6c1281fca5257F0B5,0x4faCFc8Cdf8c3C45dd0C16Cb50F44b45a2cF7466,0xc5ac1E360cfe7429c7694fF344843212c71f628b --network=mainnet
```

Unfortunately, the script will still fail due to the fact that the deployed bytecode is different on mainnet, but at least we are getting the orders. Proper testing can be done after Bens PR is finished.

### CODE REVIEW

**Comment by bh2smith** [comment]
If you rebase this on my open PR, you will find that the bytecode verification has been disabled!

**Comment by bh2smith** [comment]
Failing test appears to be ` 1) throws if master's fallback handler changed`

File: scripts/utils/verify_scripts.js [hunk 1]
```diff
@@ -90,7 +88,8 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
     log("- Verify unchanged fallback handler")
     const defaultFallbackHandler = getFallbackHandler(GnosisSafe.address)
     await Promise.all(
-      bracketAddresses.concat(masterAddress).map(async (safeAddress) => {
+      //Todo: additionally the fallback handler of the masterAddress could be checked .concat(masterAddress)
+      bracketAddresses.map(async (safeAddress) => {
```
Comment (fedgiac) [inline scripts/utils/verify_scripts.js:92]: Changes in question.

**Comment by josojo** [comment]
> The tests correctly fail because the check on whether master's fallback handler has been modified has been removed. This change is unrelated to the rest of the PR and I don't see why it would be needed in the first place.

I changed it because otherwise the testplan would fail.
```
npx truffle exec scripts/verify_correctness.js --masterSafe=0x5CC3a4C846aB8b0B78a729b8B1a50e2E0BB66740 --brackets=0x7c46953a4f0b404e21EBC0d69883aDAef5AAa5C5,0x41B77D6fAB22F67595dAF54C21b8Fc1b3812b528,0xd349236D1333F68761bbE263Af02D57B4aC928da,0x0F5D2E870f5d638D460F128530E278f0C7e4E2C6,0x5C111aFFdA2b8de51DDB7a2111111d70Ee494995,0x10aDB8e1FcDAe35123A7587e9A0E8e18d5445DD0,0xF8a02315A2eDf95d28AF945264434371fBFd43F8,0xeE223cF4292ac20533e7A0b81c1D4FEeD6eCf38C,0xe31F1C4e3B9699b971b7D4e7a8FC4eF0E39a70F9,0x2A0E02dd94c2b34ba8f94eB00807A3ABa34e62cD,0x5a80263780898C6AF01A6d57FA717aB8E3640bc3,0x47628ad01892a06cA574e45dfe8Bfb69a1cB1971,0x1c97057713C2C56599f694A920b060f1da5D94A7,0x364Ad04e974A7CbCbA5C07A867edafff5Dd12586,0xda43a32F07A41B0e1b6b53a62B593C577c65C81a,0xc31997f8e495a51E3D2346674d0Edb6804dEc1D3,0xd07C67bDf9454B6Ea26b2f5E1160A5EEE1b0f299,0xc8E1275babc318721EAa73a6c1281fca5257F0B5,0x4faCFc8Cdf8c3C45dd0C16Cb50F44b45a2cF7466,0xc5ac1E360cfe7429c7694fF344843212c71f628b --network=mainnet

> Warning: possible unsupported (undocumented in help) command line option: --masterSafe,--brackets
Using network 'mainnet'.

Warning: unable to retrieve price information on dex.ag. The server returns:
> Token not listed, please see /tokens-list-full for a list of available tokens.
Warning: unable to retrieve price information on dex.ag. The server returns:
> Token not listed, please see /tokens-list-full for a list of available tokens.
Warning: master safe owner verification skipped
- Verify that brackets are owned solely by masterSafe
- Verify that masterCopy of brackets is the known masterCopy
- Verify absence of modules
- Verify unchanged fallback handler
AssertionError [ERR_ASSERTION]: Fallback handler of Safe 0x5CC3a4C846aB8b0B78a729b8B1a50e2E0BB66740 changed
    at Promise.all.bracketAddresses.concat.map (/Users/alexherrmann/gnosis/dex-liquidity-provision/scripts/utils/verify_scripts.js:93:16)
    at process._tickCallback (internal/process/next_tick.js:68:7)

```
I think it is due to the update of the gnosis-safe contract for the master-safe. This needs to be fixed in another PR.

**Comment by bh2smith** [comment]
@josojo  I know how to fix this. Has to do the same recently in a different PR. See #255

File: scripts/utils/verify_scripts.js [hunk 2]
```diff
@@ -110,19 +109,19 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
...
-    // Fetch all token infos(decimals, symbols etc) and prices upfront for the following verification
-    const relevantOrders = auctionElementsDecoded.filter((order) => bracketTraderAddresses.includes(order.user.toLowerCase()))
     const BatchExchangeContract = Contract(require("@gnosis.pm/dex-contracts/build/contracts/BatchExchange"))
     BatchExchangeContract.setProvider(web3.currentProvider)
     const exchange = await BatchExchangeContract.deployed()
 
+    // Fetch all token infos(decimals, symbols etc) and prices upfront for the following verification
+    const ordersObjects = await Promise.all(
+      bracketTraderAddresses.map(async (bracketAddress) =>
+        exchangeUtils.decodeOrders(await exchange.getEncodedUserOrders.call(bracketAddress))
```
Comment (bh2smith) [inline scripts/utils/verify_scripts.js:120]: It appears the internal call `await exchange.getEncodedUserOrders.call(bracketAddress)` is returning null and `decodeOrders` can't handle it. I will look into the contracts repo about handling this.

Comment (bh2smith) [reply scripts/utils/verify_scripts.js:120]: Ok, so it turns out your problem is here orders object is `[ undefined ]` which is then looped over later. There was an import statement which corrected the error with `decodeOrders`, but then later you are trying to access `order.sellToken` which wont work when `order` is `undefined`. 

I think that the construction of `orderObjects` here should be revisited.

I will push my change to the import statements.

File: scripts/utils/verify_scripts.js [hunk 3]
```diff
@@ -110,19 +110,19 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
...
+      )
+    )
+    const relevantOrders = [].concat(...ordersObjects)
```
Comment (bh2smith) [inline scripts/utils/verify_scripts.js:124]: Like here, we could use
```
const relevantOrders = [].concat(...ordersObjects).filter(x => !!x)
```

Comment (bh2smith) [reply scripts/utils/verify_scripts.js:124]: I'll give it a try right here.
```suggestion
    const relevantOrders = [].concat(...ordersObjects).filter(x => !!x)
```

**Comment by josojo** [comment]
Thank you so much @bh2smith ! Looks great. I think it is ready to merge

**Pull request merged**